views/default/input/select.php

```diff
@@ -59,14 +63,22 @@
 			'selected' => in_array((string)$opt_value, $value),
 		);
 
+		if (!empty($options_attributes[$opt_value])) {
+			$option_attrs = array_merge($option_attrs, $options_attributes[$opt_value]);
+		}
+
 		$options_list .= elgg_format_element('option', $option_attrs, $option);
 	}
 } else {
 	if (is_array($options)) {
-		foreach ($options as $option) {
+		foreach ($options as $opt_key => $option) {
 
 			$option_attrs = ['selected' => in_array((string)$option, $value)];
 
+			if (!empty($options_attributes[$opt_key])) {
+				$option_attrs = array_merge($option_attrs, $options_attributes[$opt_key]);
```

Comment:
Are we intentionally allowing overwriting `selected` attribute by passing it with the attributes array?